server/lib/transactions.js

```diff
@@ -242,3 +247,126 @@ export async function sum(where) {
   const result = await models.Transaction.findOne({ attributes, where });
   return result.dataValues.total;
 }
+
+const kindStrings = {
+  ADDED_FUNDS: `Added Funds`,
+  BALANCE_TRANSFER: `Balance Transfer`,
+  CONTRIBUTION: `Contribution`,
+  EXPENSE: `Expense`,
+  HOST_FEE: `Host Fee`,
+  HOST_FEE_SHARE: `Host Fee Share`,
+  HOST_FEE_SHARE_DEBT: `Host Fee Share Debt`,
+  PAYMENT_PROCESSOR_COVER: `Cover of Payment Processor Fee`,
+  PLATFORM_TIP: `Platform Tip`,
+  PLATFORM_TIP_DEBT: `Platform Tip Debt`,
+  PREPAID_PAYMENT_METHOD: `Prepaid Payment Method`,
+};
+
+export async function generateDescription(transaction, { req = null, full = false } = {}) {
+  let baseString = 'Transaction',
+    debtString = '',
+    tierString = '',
+    expenseString = '',
+    fromString = '',
+    toString = '';
+
+  if (transaction.isRefund && transaction.RefundTransactionId) {
+    const refundedTransaction = await (req
+      ? req.loaders.Transaction.byId.load(transaction.RefundTransactionId)
+      : models.Transactions.findByPk(order.RefundTransactionId));
+    if (refundedTransaction) {
+      const refundedTransactionDescription = await generateDescription(refundedTransaction, { req, full });
+      return `Refund of \`${refundedTransactionDescription}\``;
+    }
+  }
+
+  let order, expense, subscription, tier;
+
+  if (transaction.OrderId) {
+    order = await (req
+      ? req.loaders.Order.byId.load(transaction.OrderId)
+      : models.Orders.findByPk(transaction.OrderId));
+  }
+
+  if (kindStrings[transaction.kind]) {
+    baseString = kindStrings[transaction.kind];
+  }
+
+  if (transaction.kind === CONTRIBUTION) {
+    if (order && order.TierId) {
+      tier = await (req ? req.loaders.Tier.byId.load(order.TierId) : models.Tiers.findByPk(order.TierId));
+    }
+    if (order && order.SubscriptionId) {
+      subscription = await (req
+        ? req.loaders.Subscription.byId.load(order.SubscriptionId)
+        : models.Subscriptions.findByPk(order.SubscriptionId));
+    }
+    if (subscription && subscription.interval === 'month') {
+      baseString = `Monthly contribution`;
+    } else if (subscription && subscription.interval === 'year') {
+      baseString = `Yearly contribution`;
+    } else if (tier && tier.type === TICKET) {
+      baseString = `Registration`;
+    }
+    if (tier) {
+      tierString = ` (${tier.name})`;
+    }
```

Comment:
```suggestion
    if (order?.TierId) {
      const tier = await (req ? req.loaders.Tier.byId.load(order.TierId) : models.Tiers.findByPk(order.TierId));
      tierString = ` (${tier.name})`;
    }
```

Reply:
to be on the defensive side, it's possible that `tier` is null. If the tier was deleted.